CONTRIBUTING.md

```diff
@@ -0,0 +1,59 @@
+Package Contribution Guidelines
+===============================
+
+If you are submitting a new Spack package, please observe the
+following standards:
```

Comment:
I would avoid the use of the word *standards* because that implies a whole slew of process that scares people away.  How about using the word "conventions" and emphasizing that this helps the Spack community function by handling common things consistently?